src/components/pages/maintenance/ruleDetails/ruleDetails.js

```diff
@@ -132,17 +134,24 @@ export class RuleDetails extends Component {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
 
-  // TODO: Handle error and isPending states
-  updateAlertStatus = (selectedAlerts, status) =>
+  // TODO: Handle error states
+  updateAlertStatus = (selectedAlerts, status) => {
+    this.setState({
+      isAlertStatusPending: true,
+      alertContextBtns: this.getAlertContextBtns(selectedAlerts, true)
+    });
+
     this.subscriptions.push(
       Observable.from(selectedAlerts)
         .flatMap(({ id }) => TelemetryService.updateAlertStatus(id, status))
         .toArray() // Use toArray to wait for all calls to succeed
         .subscribe(() => {
+          this.setState({ isAlertStatusPending: false });
```

Comment:
Set the pending in the `onComplete`. This subscription doesn't have error handling yet so if an error occurs the pending flag will never be unset.